src/providers/LocalizationProvider.tsx

```diff
@@ -59,6 +67,19 @@ export default function LocalizationProvider({
     fetchStrings();
   }, [locale, setLoadedStringsForLocale]);
 
+  useEffect(() => {
+    const updateUserLocale = async () => {
+      const oldProfile = (await getUser())?.user;
+
+      if (oldProfile) {
+        const newProfile = { ...oldProfile, locale };
+        await updateUserProfile(newProfile, true);
```

Comment:
may need to call `reloadData` to update the user data in user provider.
reloadData is available in useUser();